sql/sql_class.cc

```diff
@@ -898,6 +898,8 @@ extern "C" void thd_report_innodb_stat(THD *thd, unsigned long long trx_id,
                                        enum mysql_trx_stat_type type,
                                        unsigned long long       value)
 {
+  if (thd_log_slow_verbosity(thd) && (thd->innodb_trx_id != trx_id)) thd->innodb_trx_id = 0;
+
```

Comment:
Few questions

1. Could you please elaborate why this assertion happes only with `log_slow_verbosity=innodb` or `log_slow_verbosity=full`?
2. What is the reason for the different values of `thd->innodb_trx_id` and `trx_id`?
3. Are there any side-effects when the function `thd_report_innodb_stat()` is called from other paths when `log_slow_verbosity` is set.?
```
Functions calling this function: thd_report_innodb_stat

  File      Function               Line
0 trx0trx.h start_io_read          1465 thd_report_innodb_stat(trx.mysql_thd, trx.id, MYSQL_TRX_STAT_IO_READ_BYTES,
1 trx0trx.h end_io_read            1480 thd_report_innodb_stat(trx.mysql_thd, trx.id,
2 trx0trx.h bump_io_read           1491 thd_report_innodb_stat(trx.mysql_thd, trx.id, MYSQL_TRX_STAT_IO_READ_BYTES,
3 trx0trx.h bump_innodb_enter_wait 1502 thd_report_innodb_stat(trx.mysql_thd, trx.id,
4 trx0trx.h stop_lock_wait         1509 thd_report_innodb_stat(trx.mysql_thd, trx.id, MYSQL_TRX_STAT_LOCK_WAIT_USECS,
5 trx0trx.h inc_page_get           1526 thd_report_innodb_stat(trx.mysql_thd, trx.id, MYSQL_TRX_STAT_ACCESS_PAGE_ID,
```

Also, if possible, could you please also add an MTR testcase for the same?